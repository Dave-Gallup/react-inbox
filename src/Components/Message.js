import React, { Component } from 'react';
import Label from './Label.js';

class Message extends Component{

  constructor(props){
    super(props);

    this.onCheckboxClick = this.onCheckboxClick.bind(this);
    this.onStarClick = this.onStarClick.bind(this);
  }


  onCheckboxClick(){
    var change = {};
    change.selected = !this.props.selected;
    this.props.updateState(change, this.props.id);
  }

  onStarClick(){
    var change = {};
    change.starred = !this.props.starred;
    this.props.updateState(change, this.props.id);
  }


  render(){
    var star = this.props.starred ? 'fa-star' : 'fa-star-o';
    var read = this.props.read ? 'read' : 'unread';
    var selectedBox = this.props.selected ? 'checked' : '';
    var selectedBg = this.props.selected ? 'selected' : '';




    return (
      <div className={"row message " + read + " " + selectedBg} >
        <div className="col-xs-1">
          <div className="row">
            <div className="col-xs-2">
              <input type="checkbox" checked={"" + selectedBox} onClick={this.onCheckboxClick}/>
            </div>
            <div className="col-xs-2">
              <i className={"star fa " + star} onClick={this.onStarClick}></i>
            </div>
          </div>
        </div>
        <div className="col-xs-11 text-left" onClick={this.onCheckboxClick}>
          {this.props.labels.map((el, i) => <Label label={el} key={i}/>)}
          <a>
            {this.props.subject}
          </a>
        </div>
      </div>
    );
  }

}


export default Message;