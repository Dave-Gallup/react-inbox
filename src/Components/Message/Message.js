import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Label from '../Label/Label.js';
import Body from '../Body/Body.js';
import {
  toggleSelected,
  toggleStarred,
  setRead,
} from '../../actions/actions';
import {BrowserRouter as Router, Route} from 'react-router-dom';

class Message extends Component{

  constructor(props){
    super(props);

    this.onSelectedChange = this.onSelectedChange.bind(this);
    this.onStarredClick = this.onStarredClick.bind(this);
    this.onSubjectClick = this.onSubjectClick.bind(this);
  }

  onSelectedChange(){
    this.props.toggleSelected(this.props.id);
  }

  onStarredClick(){
    this.props.toggleStarred(this.props.id);
  }

  onSubjectClick(){
    this.props.setRead([this.props.id], true);
    if(window.location.pathname === `/message/${this.props.id}`){
      window.location.href='http://localhost:3000/';
    }
    else{
      window.location.href=`http://localhost:3000/message/${this.props.id}`;
    }
  }


  render(){
    return (
      <div  className={`row message ${this.props.read?'':'un'}read ${this.props.selected?'selected':''}`}>
        <div className="col-xs-1">
          <div className="row">
            <div className="col-xs-2">
              <input type="checkbox" checked={this.props.selected ? 'checked' : ''} onChange={this.onSelectedChange}/>
            </div>
            <div className="col-xs-2">
              <i className={`star fa ${this.props.starred?'fa-star':'fa-star-o'}`} onClick={this.onStarredClick}/>
            </div>
          </div>
        </div>
        <div className="col-xs-11 text-left" >

          <a onClick={this.onSubjectClick}>
            {this.props.labels.map((el, i) => <Label key={el} messageId={this.props.id} labelId={i}/>)}
            {this.props.subject}
          </a>

        </div>
        <Router>
          <Route path="/message/:id(\d+)" render={({match})=>(
            parseInt(match.params.id, 10) === this.props.id ? <Body id={this.props.id} /> : null
          )} />
        </Router>
      </div>
    );
  }



}

const mapStateToProps = (state, ownProps) => {

  const subject  = state.messages.messageMap[ownProps.id].subject;
  const labels   = state.messages.messageMap[ownProps.id].labels;
  const starred  = state.messages.messageMap[ownProps.id].starred;
  const read     = state.messages.messageMap[ownProps.id].read;
  const selected = state.messages.messageMap[ownProps.id].selected;
  const body     = state.messages.messageMap[ownProps.id].body;
  return {
    subject,
    labels,
    starred,
    read,
    selected,
    body
  }
};

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSelected,
  toggleStarred,
  setRead
},
dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Message);


