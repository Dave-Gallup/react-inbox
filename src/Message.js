import React, { Component } from 'react';
import './index.css';
import Label from './Label.js';

class Message extends Component{

  constructor(props){
    super(props);
  }

  render(){
    var star = this.props.starred ? 'fa-star' : 'fa-star-o';
    var read = this.props.read ? 'read' : 'unread';

    return (
      <div className={"row message " + read}>
        <div className="col-xs-1">
          <div className="row">
            <div className="col-xs-2">
              <input type="checkbox"/>
            </div>
            <div className="col-xs-2">
              <i className={"star fa " + star}></i>
            </div>
          </div>
        </div>
        <div className="col-xs-11 text-left">
          {this.props.labels.map((el) => <Label label={el} />)}
          <a href="#">
            {this.props.subject}
          </a>
        </div>
      </div>
    );
  }

}


export default Message;