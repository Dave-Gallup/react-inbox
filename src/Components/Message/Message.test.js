import React from 'react';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Message, mapStateToProps } from './Message';

describe('Message', () => {
  it('should render without any props other than empty labels defined', () => {
    const component = shallow(
      <Message labels={[]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  it('should render with starred=true prop passed in', () => {
    const component = shallow(
      <Message starred={true} labels={[]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  it('should render with selected=true prop passed in', () => {
    const component = shallow(
      <Message selected={true} labels={[]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })

  it('should render with read=true prop passed in', () => {
    const component = shallow(
      <Message read={true} labels={[]}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  })


  it('mapStateToProps', () => {
    const state = {
      messages: {
        messageList: [1,2],
        messageMap:{
          1:{
            subject: 'test subject',
            labels:[
              'test-label1',
              'test-label2'
            ],
            starred: true,
            read: false,
            selected: false,
            body: 'This is the test body.'
          },
          2:{
            subject: 'another test subject',
            labels:[
              'test-label2',
              'test-label4'
            ],
            starred: false,
            read: true,
            selected: true,
            body: 'Second test body.'
          }
        }
      }
    }

    const ownProps = {
      id: 1
    }

    const expectedResult = {
      subject: 'test subject',
      labels:[
        'test-label1',
        'test-label2'
      ],
      starred: true,
      read: false,
      selected: false,
      body: 'This is the test body.'
    }

    expect(mapStateToProps(state, ownProps)).toEqual(expectedResult)
  })



    //
    // component.find('[name="toggle-preview"]').simulate('click');
    // expect(toJson(component)).toMatchSnapshot();
});
