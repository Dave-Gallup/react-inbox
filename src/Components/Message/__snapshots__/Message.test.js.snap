// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render with read=true prop passed in 1`] = `
<div
  className="row message read "
>
  <div
    className="col-xs-1"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-2"
      >
        <input
          checked=""
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="col-xs-2"
      >
        <i
          className="star fa fa-star-o"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="col-xs-11 text-left"
  >
    <a
      onClick={[Function]}
    />
  </div>
  <Route
    path="/message/:id(\\\\d+)"
    render={[Function]}
  />
</div>
`;

exports[`Message should render with selected=true prop passed in 1`] = `
<div
  className="row message unread selected"
>
  <div
    className="col-xs-1"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-2"
      >
        <input
          checked="checked"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="col-xs-2"
      >
        <i
          className="star fa fa-star-o"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="col-xs-11 text-left"
  >
    <a
      onClick={[Function]}
    />
  </div>
  <Route
    path="/message/:id(\\\\d+)"
    render={[Function]}
  />
</div>
`;

exports[`Message should render with starred=true prop passed in 1`] = `
<div
  className="row message unread "
>
  <div
    className="col-xs-1"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-2"
      >
        <input
          checked=""
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="col-xs-2"
      >
        <i
          className="star fa fa-star"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="col-xs-11 text-left"
  >
    <a
      onClick={[Function]}
    />
  </div>
  <Route
    path="/message/:id(\\\\d+)"
    render={[Function]}
  />
</div>
`;

exports[`Message should render without any props other than empty labels defined 1`] = `
<div
  className="row message unread "
>
  <div
    className="col-xs-1"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-2"
      >
        <input
          checked=""
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="col-xs-2"
      >
        <i
          className="star fa fa-star-o"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="col-xs-11 text-left"
  >
    <a
      onClick={[Function]}
    />
  </div>
  <Route
    path="/message/:id(\\\\d+)"
    render={[Function]}
  />
</div>
`;
